pkg/helper/comparators.go

```diff
@@ -57,6 +92,28 @@ func annotationsChanged(old, new *corev1.PodTemplateSpec, report *ChangeReport)
 	return false
 }
 
+func assignationChanged(old, new *corev1.PodTemplateSpec, report *ChangeReport) bool {
+	if !equality.Semantic.DeepDerivative(old.Spec.NodeSelector, new.Spec.NodeSelector) {
```

Comment:
`DeepDerivative` normally takes the "new" param as a first argument and the "old" as the second; and it checks for stuff in "new" that wasn't  in "old", but not the other way around. That's typically useful for annotations/labels, when other parts of the kube infra might add their own labels / annotations on pods and we don't want to interfere with that.

So here I believe either we want to use `DeepDerivative` but then we should change the arguments order, or perhaps we actually want `DeepEqual` here ?

Reply:
I gave a try with `DeepEqual` here and got a infinite reconcile loop. It seemed to work fine with `DeepDerivative` but I'll give another try inverting the two args to see if I get differences.

Reply:
I have addressed your feedback and simplified the code in this commit: https://github.com/netobserv/network-observability-operator/pull/569/commits/4178913ec45ffaab1183d4941bf84583389f61bc

Thanks !